src/concurrency.md

```diff
@@ -33,6 +34,40 @@ fn main() {
 }
 ```
 
+[ex-rayon-any-all]: #ex-rayon-any-all
+<a name="ex-rayon-any-all"></a>
+## Check the elements of an array against a predicate in parallel
+
+[![rayon-badge]][rayon] [![cat-concurrency-badge]][cat-concurrency]
+
+The `rayon` provides a [`ParallelIterator`] trait which includes parallelized implementations of many of the methods from the standard library's [`Iterator`] trait.
+
+This example demonstrates using the [`any`] and [`all`] methods. [`any`] checks in parallel whether any element of the iterator matches the predicate, and returns as soon as one is found. [`all`] checks in parallel whether all elements of the iterator match the predicate, and returns as soon as a non-matching element is found.
+
+```rust
+extern crate rayon;
+
+use rayon::prelude::*;
+
+fn main() {
```

Comment:
There seems to be a mix of examples using a `main()` function and a those using `error_chain`. I'm not sure the Contributing guide is clear on this either, as it does not explicitly state contributions should use `error_chain`. On the other hand, there are no errors we need to worry about here, so maybe using `error_chain` unnecessarily complicates the examples.

@budziq what are your thoughts here? I am somewhat partial to using a consistent approach across examples.

Reply:
@j-haj Thanks for taking this on :+1: . 
You are right our [note about error handling](https://rust-lang-nursery.github.io/rust-cookbook/about.html#a-note-about-error-handling) does not specifically state what is the correct course of action if there are not errors to handle. 

The approach is as follows. If there are no errors to handle, there is no reason to increase the complexity of the example with `run` and `error_chain` and we just use plain `main`. We might be interested in adding this information to our contributing and error handling sections

Reply:
I was also wondering about this. Thanks!